src/core/plugin_manager.py

```
@@ -375,9 +377,19 @@ async def _download_and_install_plugin(self, plugin_info: PluginMarketplaceEntry
     async def _download_file(self, url: str, dest_path: Path):
         """Download a file from URL."""
         try:
-            with urlopen(url) as response:
-                with open(dest_path, "wb") as f:
-                    f.write(response.read())
+            # Enforce URL scheme validation (SSRF protection)
+            parsed_url = urlparse(url)
+            if parsed_url.scheme not in ('http', 'https'):
+                raise SecurityError(f"Invalid URL scheme: {parsed_url.scheme}. Only http and https are allowed.")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)